src/handleApps.test.ts

```diff
@@ -174,6 +175,49 @@ describe(`handleApps`, () => {
       "public, max-age=31536000, immutable"
     );
   });
+
+  it(`can retrieve a file from s3`, async () => {
+    const orgSettings: RecursivePartial<OrgSettings> = {
+      orgExists: true,
+      staticFiles: {
+        microfrontendProxy: {
+          environments: {
+            __main__: {
+              useFoundryHosting: false,
+              customHost: "s3://example",
+            },
+          },
+        },
+      },
+    };
+
+    (global.MAIN_KV as MockCloudflareKV).mockKv({
+      "org-settings-walmart": orgSettings,
+    });
+
+    sendMock.mockReturnValueOnce(
+      Promise.resolve({
+        Body: "var a = 1",
+        ContentType: "application/javascript",
+      })
+    );
+
+    response = await handleApps(
+      new Request("https://cdn.baseplate.cloud/walmart/apps/example.js"),
+      {
+        orgKey: "walmart",
+        customerEnv: "__main__",
+        pathParts: ["example.js"],
+      }
+    );
+    let responseBody = await response.text();
+
+    expect(response.ok).toBe(true);
+    expect(response.headers.get("content-type")).toEqual(
+      "application/javascript"
+    );
+    expect(responseBody).toEqual("var a = 1");
```

Comment:
I'd add one more assertion here, to verify it actually called S3 to get the file:

```ts
expect(sendMock).toHaveBeenCalled();
```